server/channels/api4/channel_test.go

```diff
@@ -4074,37 +4212,44 @@ func TestAddChannelMember(t *testing.T) {
 
 	otherUser := th.CreateUser()
 	otherChannel := th.CreatePublicChannel()
-	client.Logout(context.Background())
-	client.Login(context.Background(), user2.Id, user2.Password)
+
+	_, err = client.Logout(context.Background())
+	require.NoError(t, err)
+	_, _, err = client.Login(context.Background(), user2.Email, user2.Password)
+	require.NoError(t, err)
 
 	_, resp, err = client.AddChannelMember(context.Background(), publicChannel.Id, otherUser.Id)
 	require.Error(t, err)
-	CheckUnauthorizedStatus(t, resp)
+	CheckNotFoundStatus(t, resp)
 
 	_, resp, err = client.AddChannelMember(context.Background(), privateChannel.Id, otherUser.Id)
 	require.Error(t, err)
-	CheckUnauthorizedStatus(t, resp)
+	CheckNotFoundStatus(t, resp)
 
 	_, resp, err = client.AddChannelMember(context.Background(), otherChannel.Id, otherUser.Id)
 	require.Error(t, err)
-	CheckUnauthorizedStatus(t, resp)
+	CheckForbiddenStatus(t, resp)
 
-	client.Logout(context.Background())
-	client.Login(context.Background(), user.Id, user.Password)
+	_, err = client.Logout(context.Background())
+	require.NoError(t, err)
+	_, _, err = client.Login(context.Background(), user.Email, user.Password)
+	require.NoError(t, err)
 
 	// should fail adding user who is not a member of the team
 	_, resp, err = client.AddChannelMember(context.Background(), otherChannel.Id, otherUser.Id)
 	require.Error(t, err)
-	CheckUnauthorizedStatus(t, resp)
+	CheckNotFoundStatus(t, resp)
 
-	client.DeleteChannel(context.Background(), otherChannel.Id)
+	_, err = client.DeleteChannel(context.Background(), otherChannel.Id)
+	require.NoError(t, err)
 
-	// should fail adding user to a deleted channel
+	// Adding user to a deleted channel is fine
 	_, resp, err = client.AddChannelMember(context.Background(), otherChannel.Id, user2.Id)
-	require.Error(t, err)
-	CheckUnauthorizedStatus(t, resp)
+	require.NoError(t, err)
+	CheckCreatedStatus(t, resp)
```

Comment:
Here, the business logic check is wrong. Users _can_ be added to deleted channels.